services/s3/s3.go

```diff
@@ -286,10 +295,32 @@ func (s *S3) readerForRange(object *Object, br ByteRange) (io.Reader, *awserrors
 		}
 
 		readers = append(readers, io.NewSectionReader(f, bytesUntilStart, bytesToReadFromThisChunk))
+		bytesUntilEnd -= (bytesUntilStart + bytesToReadFromThisChunk)
 		bytesUntilStart = 0
-		bytesUntilEnd -= int64(bytesToReadFromThisChunk)
+		readLength += bytesToReadFromThisChunk
+	}
+	if bytesUntilStart > 0 {
+		// For whatever reason, the AWS sdk requires that this error message is formatted correctly.
+		errorObj := InvalidRangeError{
```

Comment:
lolol, this seems like some poor AWS engineer did a weird thing and then they baked it into the protocol :)